sympy/concrete/tests/test_sums_products.py

```diff
@@ -1300,11 +1302,11 @@ def test_empty_sequence():
     assert Sum(x, (x, -oo, oo), (y, 1, 0)).doit() == 0
     assert Sum(x, (y, 1, 0), (x, -oo, oo)).doit() == 0
 
-
+@XFAIL
 def test_issue_8016():
-    k = Symbol('k', integer=True)
+    k = Symbol('k', integer=True, nonnegative=True)
     n, m = symbols('n, m', integer=True, positive=True)
-    s = Sum(binomial(m, k)*binomial(m, n - k)*(-1)**k, (k, 0, n))
+    s = Sum(binomial(m, k)*binomial(m, n - k)*(-1)**k, (k, 0, n)) # We don't know if m > (n - k)
```

Comment:
@smichr  could you take a look at this ?

Reply:
Sum should confer the subsets of attributes of the limits onto the variable of summation. So even if k were vanilla, it should get a nonnegative and integer attributes from the given limit from 0 to a positive integer. I'll look.